src/build.rs

```diff
@@ -122,17 +136,15 @@ impl Builder {
             }
         }
     }
+}
 
-    fn crate_names(&self) -> Vec<String> {
-        let metadata = match cargo_metadata::metadata_deps(None, true) {
-            Ok(metadata) => metadata,
-            Err(_) => return Vec::new(),
-        };
-
-        metadata
-            .packages
-            .into_iter()
-            .map(|p| p.name)
-            .collect()
-    }
+fn crate_names() -> impl Iterator<Item = String> {
```

Comment:
Making this a free function rather than method is fine, but the other refactoring I think is not worth doing - iiuc, it adds some complexity only for the sake of removing a single allocation, which I think is unlikely to be significant.

Reply:
I don't think it really adds complexity, just a "what's happening here?" with the mapping of an empty vec.

We can replace it with: 

```rust
fn crate_names() -> impl Iterator<Item = String> {
    cargo_metadata::metadata_deps(None, true)
        .map(|metadata| metadata.packages)
        .unwrap_or_default()
        .map(|package| package.name)
}
```